packages/paste-core/components/button/src/types.ts

```diff
@@ -82,16 +98,18 @@ export interface DirectButtonProps extends HTMLPasteProps<"button"> {
    */
   type?: ButtonTypes;
   /**
+   * The different appearance variants for a button. [Avoid using link variants when possible.](https://paste.twilio.design/components/button#button-vs-anchor-link)
+   *
    * @default 'primary'
    */
   variant: ButtonVariants;
 }
 
-type BaseVariantsButtonProps = {
-  variant?: ButtonBaseVariants;
-};
-type ResetVariantButtonProps = Omit<BoxStyleProps, "size"> & {
-  variant?: ButtonResetVariant;
+type VariantsButtonProps = {
+  /**
+   * The different appearance variants for a button. [Avoid using link variants when possible.](https://paste.twilio.design/components/button#button-vs-anchor-link)
+   */
+  variant?: ButtonBaseVariants | ButtonResetVariant;
```

Comment:
We had two types of variables here however we only used it in one place. Restructuring the type this way allows for guidance on the property to be included.

Reply:
This actually leads to an error, as this typing was intentional